### PR TITLE
deepin.deepin-voice-note: init at 5.10.22

### PR DESCRIPTION
**Pull request by wineee**
###### Description of changes
https://github.com/linuxdeepin/dde-nixos/issues/9

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wineee** [comment]
@ofborg build deepin.deepin-voice-note

File: pkgs/desktops/deepin/apps/deepin-voice-note/default.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  ]);
+
+  cmakeFlags = [ "-DVERSION=${version}" ];
+
+  NIX_CFLAGS_COMPILE = "-I${dde-qt-dbus-factory}/include/libdframeworkdbus-2.0";
+
+  # qt5integration must be placed before qtsvg in QT_PLUGIN_PATH
+  qtWrapperArgs = [
+    "--prefix QT_PLUGIN_PATH : ${qt5integration}/${qtbase.qtPluginPrefix}"
+    "--prefix GST_PLUGIN_SYSTEM_PATH_1_0 : ${gstPluginPath}"
```
Comment (NickCao) [inline pkgs/desktops/deepin/apps/deepin-voice-note/default.nix:68]: According to `doc/languages-frameworks/gnome.section.md`, Setup hooks of `gst_all_1.gstreamer` will populate the `GST_PLUGIN_SYSTEM_PATH_1_0`, so no need to manually construct it here?

Comment (wineee) [reply pkgs/desktops/deepin/apps/deepin-voice-note/default.nix:68]: We need `wrapGAppsHook` to set `GST_PLUGIN_SYSTEM_PATH_1_0`, but this application only use `wrapQtAppsHook`

Comment (NickCao) [reply pkgs/desktops/deepin/apps/deepin-voice-note/default.nix:68]: `GST_PLUGIN_SYSTEM_PATH_1_0` is used by `wrapGAppsHook`, but set by `gstreamer` setup hooks.

Comment (NickCao) [reply pkgs/desktops/deepin/apps/deepin-voice-note/default.nix:68]: `--prefix GST_PLUGIN_SYSTEM_PATH_1_0 : $GST_PLUGIN_SYSTEM_PATH_1_0` should work.

**Comment by NickCao** [comment]
@ofborg build deepin.deepin-voice-note

**Review by NickCao — APPROVED** [review]

**Pull request merged**